### PR TITLE
hotfix/LLVM linker flags for CPURuntime 

### PR DESCRIPTION
**Pull request by dendisuhubdy**
adding correct linking flags for CPURuntime CMakeLists.txt. Regarding #1505

### CODE REVIEW

**Comment by nadavrot** [comment]
@compnerd

File: lib/Importer/CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@ if(NOT TARGET onnx_proto)
   add_subdirectory(${GLOW_THIRDPARTY_DIR}/onnx EXCLUDE_FROM_ALL build)
 endif()
 
-find_package(Protobuf REQUIRED)
+FIND_PACKAGE(Protobuf REQUIRED)
```
Comment (lygstate) [inline lib/Importer/CMakeLists.txt]: No need uppercase

File: lib/Backends/CPU/CMakeLists.txt [hunk 2]
```diff
@@ -30,13 +30,8 @@ target_compile_options(CPURuntime
...
 # linker to merge the object files.
 set_target_properties(CPURuntime
-                      PROPERTIES
-                        RULE_LAUNCH_COMPILE
-                          "${CLANG_BIN} <DEFINES> <INCLUDES> <FLAGS> -o <OBJECT> -c <SOURCE> # "
-                        RULE_LAUNCH_LINK
-                          "${LLVM_LINK_BIN} -o ${CMAKE_BINARY_DIR}/libjit.bc <OBJECTS> # "
-                        OUTPUT_NAME
-                          libjit.bc)
+                      PROPERTIES LINK_FLAGS "-Wl,-object_path_lto"
```
Comment (compnerd) [inline lib/Backends/CPU/CMakeLists.txt]: Please dont use `-Wl,` as that prevents some characters in the invocation.  The `-object_path_lto` is specific to Darwin, so this will break non-Darwin builds.  The custom invocation is absolutely required for non-ld64 builds.

File: lib/Backends/CPU/CMakeLists.txt [hunk 3]
```diff
@@ -55,15 +50,18 @@ add_library(CPUBackend
...
                       PRIVATE
                         Base
                         CodeGen
                         Graph
                         IR
                         QuantizationBase
-                        LLVMAnalysis
+                        LLVMObject
+                        LLVMBitWriter
+						LLVMAnalysis
```
Comment (compnerd) [inline lib/Backends/CPU/CMakeLists.txt]: Are there hard tabs?  Please use spaces.

File: tests/unittests/CMakeLists.txt [hunk 4]
```diff
@@ -212,7 +212,8 @@ target_link_libraries(caffe2ImporterTest
                         Importer
                         ExecutionEngine
                         gtest
-                        testMain)
+                        testMain
+                        )
```
Comment (compnerd) [inline tests/unittests/CMakeLists.txt]: Unnecessary change?

File: tests/unittests/CMakeLists.txt [hunk 5]
```diff
@@ -225,7 +226,7 @@ target_link_libraries(onnxImporterTest
                         Importer
                         ExecutionEngine
                         gtest
-			testMain)
+			            testMain)
```
Comment (compnerd) [inline tests/unittests/CMakeLists.txt]: Please use spaces.

**Comment by bertmaher** [comment]
Fixed the llvm lib dependencies in #1831.  The libjit.bc setup has changed a lot, so this PR is probably not super relevant any longer.  Thanks for digging into the issues though!
